spec/models/form_spec.rb

```diff
@@ -8,10 +8,20 @@
 
   describe ".next_page" do
     let(:previous_page) { form.get_page("person_1_age") }
+    let(:value_check_previous_page) { form.get_page("net_income_value_check") }
 
     it "returns the next page given the previous" do
       expect(form.next_page(previous_page, case_log)).to eq("person_1_gender")
     end
+
+    it "returns the previuos page if the current page is a value check page with a `No` answer and the page is routed to" do
```

Comment:
Typo on `previous`.

Would also be good to add scenarios for:

- Previous page is a value check page with "Yes" answer
- Previous page is not a value check page

Reply:
🤦 
Previous page is a value check page with "Yes" answer - Yup!
Previous page is not a value check page but is routed to/Previous page is not a value check page and is not routed to - we wouldn't call previous_page in these scenarios tho. It would go down the route of normal next page  (?)

Reply:
although the first scenario we wouldn't call previous_page either, but I guess it checks that new if statement

Reply:
I guess it's still useful in a sort of "defensive programming" sense? i.e. if the next person misunderstands when that method can/should be called and tries to use it for that scenario what should happen (perhaps it should raise an error if it considers that incorrect usage?).

Possibly veering into overkill territory here though so happy to merge as is.

Reply:
Added a few more

Reply:
![chefkiss](https://user-images.githubusercontent.com/5101747/160841553-d1d86bca-edde-4ae4-9e9b-5cf6b05bf40d.png)